source/acceptance_test.go

```diff
@@ -0,0 +1,234 @@
+		t.Fatal(err)
+	}
+	sdk.AcceptanceTest(t, AcceptanceTestDriver{
+		Config: AcceptanceSourceTestDriverConfig{
+			Connector: sdk.Connector{
+				NewSpecification: marketo.Specification,
+				NewSource:        source.NewSource,
+			},
+			SourceConfig: getConfigs(),
+			GoleakOptions: []goleak.Option{
+				goleak.IgnoreCurrent(),
+				goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+				// external dependency - minimarketo
+				goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+				goleak.IgnoreTopFunction("net/http.(*persistConn).roundTrip"),
+				goleak.IgnoreTopFunction("net/http.setRequestCancel.func4"),
+			},
+			BeforeTest: func(t *testing.T) {
+				iterator.InitialDate = time.Now().UTC()
+				startTime = time.Now().UTC()
+			},
+			AfterTest: func(t *testing.T) {
+				t.Cleanup(func() {
+					err := cleanUp(client, startTime)
+					if err != nil {
+						t.Error(err)
+					}
+				})
+			},
+		},
+	},
+	)
+}
+
+// AcceptanceTestDriver implements sdk.AcceptanceTestDriver
+type AcceptanceTestDriver struct {
+	Config AcceptanceSourceTestDriverConfig
+}
+
+// AcceptanceSourceTestDriverConfig contains the configuration for
+// AcceptanceTestDriver.
+type AcceptanceSourceTestDriverConfig struct {
+	// Connector is the connector to be tested.
+	Connector sdk.Connector
+
+	// SourceConfig config for source
+	SourceConfig map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest func(t *testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest func(t *testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+
+	// Skip is a slice of regular expressions used to identify tests that should
+	// be skipped. The full test name will be matched against all regular
+	// expressions and the test will be skipped if a match is found.
+	Skip []string
+}
+
+func (d AcceptanceTestDriver) DestinationConfig(*testing.T) map[string]string {
+	return map[string]string{}
+}
+func (d AcceptanceTestDriver) Connector() sdk.Connector {
+	return d.Config.Connector
+}
+
+func (d AcceptanceTestDriver) SourceConfig(*testing.T) map[string]string {
+	return d.Config.SourceConfig
+}
+
+func (d AcceptanceTestDriver) BeforeTest(t *testing.T) {
+	// before test check if the test should be skipped
+	d.Skip(t)
+
+	if d.Config.BeforeTest != nil {
+		d.Config.BeforeTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) AfterTest(t *testing.T) {
+	if d.Config.AfterTest != nil {
+		d.Config.AfterTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) Skip(t *testing.T) {
+	var skipRegexs []*regexp.Regexp
+	for _, skipRegex := range d.Config.Skip {
+		r := regexp.MustCompile(skipRegex)
+		skipRegexs = append(skipRegexs, r)
+	}
+
+	for _, skipRegex := range skipRegexs {
+		if skipRegex.MatchString(t.Name()) {
+			t.Skipf("caller requested to skip tests that match the regex %q", skipRegex.String())
+		}
+	}
+}
```

Comment:
IMHO, it's better to embed SDK's test driver, instead of reimplementing all the methods.

Reply:
updated.